spec/controllers/users_controller_spec.rb

```diff
@@ -9,109 +9,38 @@
-
-        hans = User.where(email: "hans@falcon.com").first
-        expect(hans.follows?(luke)).to be_truthy
-      end
-
-      it "make the inviter follow the user" do
-        luke = Fabricate(:user)
-        invitation = Fabricate(
-          :invitation,
-          inviter: luke,
-          recipient_email: "hans@falcon.com"
-        )
-        post :create, user: {
-          email: "hans@falcon.com",
-          full_name: "Hans Solo",
-          password: "password"
-        }, invitation_token: invitation.token
-
-        hans = User.where(email: "hans@falcon.com").first
-        expect(luke.follows?(hans)).to be_truthy
-      end
-
-      it "expires the invitation upon acceptance" do
-        luke = Fabricate(:user)
-        invitation = Fabricate(
-          :invitation,
-          inviter: luke,
-          recipient_email: "hans@falcon.com"
-        )
-        post :create, user: {
-          email: "hans@falcon.com",
-          full_name: "Hans Solo",
-          password: "password"
-        }, invitation_token: invitation.token
-        expect(Invitation.first.token).to be_nil
-      end
-    end
-
-    context "email sending" do
-      it "sends out the email given valid inputs" do
-        post :create, user: Fabricate.attributes_for(
-          :user, email: "example@example.com", full_name: "The Wizard of Id"
-        )
-        expect(ActionMailer::Base.deliveries.last.to).to eq(["example@example.com"])
-      end
-
-      it "email sent out has users name" do
-        post :create, user: Fabricate.attributes_for(
-          :user, email: "example@example.com", full_name: "The Wizard of Id"
-        )
-        expect(ActionMailer::Base.deliveries.last.body).to include("The Wizard of Id")
-      end
-
-      it "email does not send given invalid inputs" do
-        post :create, user: {
-          email: "example@example.com",
-          full_name: "The Wizard of Id",
-          password: ""
-        }
-        expect(ActionMailer::Base.deliveries).to be_empty
-      end
     end
 
-    context "with invalid input" do
+    context "failed sign up" do
       before do
-        post :create, user: { password: "serenity", full_name: "Nathon Fillion" }
+        result = double(
+          :sign_up_result,
+          successful?: false,
+          error_message: "This is failure"
+        )
+        UserSignup.any_instance.should_receive(:sign_up).and_return(result)
+        post :create, user: Fabricate.attributes_for(:user)
       end
 
-      it "does not create user" do
+      it "does not create a new user record" do
         expect(User.count).to eq(0)
       end
 
-      it "renders the :new template" do
+      it "renders the new template" do
         expect(response).to render_template :new
       end
 
-      it "sets @user" do
-        expect(assigns(:user)).to be_instance_of(User)
+      it "sets the flash error message" do
+        expect(flash[:danger]).to eq("This is failure")
```

Comment:
testing that the flash message is set is rather tested `expect(flash[:danger]).to be_present` or other presence matcher.